pages/account/manage/event/[[...data]].js

```diff
@@ -89,51 +90,51 @@ export default function ManageEvent({ BASE_URL, event }) {
       putEvent = { ...putEvent, _id: event._id };
       apiUrl = `${BASE_URL}/api/account/manage/event/${event._id}`;
     }
-    const res = await fetch(apiUrl, {
-      method: "PUT",
-      headers: {
-        "Content-Type": "application/json",
-      },
-      body: JSON.stringify(putEvent),
-    });
-    const update = await res.json();
+    try {
+      await sendRequest({
+        url: apiUrl,
+        method: "PUT",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: putEvent,
+      });
+      Router.push(`${BASE_URL}/account/manage/events?success=true`);
+    } catch (err) {
+      const errMessage =
+        typeof err === "string"
+          ? err
+          : Object.keys(err)
+              .map((val) => `${val} is required`)
+              .join(", ");
 
-    if (update.message) {
-      return setShowNotification({
+      setShowNotification({
         show: true,
         type: "error",
-        message: "Event add/update failed",
-        additionalMessage: `Please check the fields: ${Object.keys(
-          update.message
-        ).join(", ")}`,
+        message: "Profile update failed",
+        additionalMessage: `Request failed! ${errMessage}`,
       });
     }
-
-    Router.push(`${BASE_URL}/account/manage/events?success=true`);
   };
 
   const deleteItem = async () => {
-    const res = await fetch(
-      `${BASE_URL}/api/account/manage/event/${event._id}`,
-      {
+    try {
+      await sendRequest({
+        url: `${BASE_URL}/api/account/manage/event/${event._id}`,
```

Comment:
I'm not sure why BASE_URL comes from server side, as it's in the config. Maybe, it can be added to api client from the config and not passed through many components

Reply:
You're right 👍